### ISSUE TITLE
How to update RegionProposalNetwork loss function in Faster RCNN? 

### ISSUE THREAD

**Issue by darian69** [pytorch/vision#7987]
Excuse me if this question is stupid, but I can't seem to figure out how to do this…

I want to update the loss function of the RPN in FasterRCNN. See these lines [here](https://github.com/pytorch/vision/blob/beb4bb706b5e13009cb5d5586505c6d2896d184a/torchvision/models/detection/generalized_rcnn.py#L104-L105), which calls the `compute_loss` function [here](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/rpn.py#L298). I want to modify the `compute_loss` function (the second link).

I’m trying to update this `compute_loss` function in my code like so:

```rpn.RegionProposalNetwork.compute_loss = custom_loss```

However, this is not working i.e. it has no effect. Any idea how to update the RPN’s loss function?

**Comment by NicolasHug** [comment]
Hi @darian69 ,

A bit ugly but you probably can patch the instance instead of trying to patch the class:

```py

from torchvision.models import resnet50, detection
backbone = resnet50()
backbone.out_channels = 2048

frcnn = detection.FasterRCNN(backbone=backbone, num_classes=10)

print(frcnn.rpn.compute_loss)

def my_custom_loss(*args, **kwargs):
    pass


frcnn.rpn.compute_loss = my_custom_loss
print(frcnn.rpn.compute_loss)
```

**Comment by darian69** [comment]
Thanks, that was it!
